drivers/usb_c/tcpc/ps8xxx.c

```diff
@@ -332,98 +315,24 @@ int ps8xxx_tcpc_transmit_data(const struct device *dev, struct pd_msg *msg)
 {
 	const struct ps8xxx_cfg *cfg = dev->config;
 
-	int reg = TCPC_REG_TX_BUFFER;
-	int rv;
-	int cnt = 4 * msg->header.number_of_data_objects;
-
-	/* If not SOP* transmission, just write to the transmit register */
-	if (msg->header.message_type >= NUM_SOP_STAR_TYPES) {
-		/*
-		 * Per TCPCI spec, do not specify retry (although the TCPC
-		 * should ignore retry field for these 3 types).
-		 */
-		return tcpci_write_reg8(
-			&cfg->bus, TCPC_REG_TRANSMIT,
-			TCPC_REG_TRANSMIT_SET_WITHOUT_RETRY(msg->header.message_type));
-	}
-
-	if (cnt > 0) {
-		reg = TCPC_REG_TX_BUFFER;
-		/* TX_BYTE_CNT includes extra bytes for message header */
-		cnt += sizeof(msg->header.raw_value);
-
-		struct i2c_msg buf[3];
-
-		uint8_t tmp[2] = {TCPC_REG_TX_BUFFER, cnt};
-
-		buf[0].buf = tmp;
-		buf[0].len = 2;
-		buf[0].flags = I2C_MSG_WRITE;
-
-		buf[1].buf = (uint8_t *)&msg->header.raw_value;
-		buf[1].len = sizeof(msg->header.raw_value);
-		buf[1].flags = I2C_MSG_WRITE;
-
-		buf[2].buf = (uint8_t *)msg->data;
-		buf[2].len = msg->len;
-		buf[2].flags = I2C_MSG_WRITE | I2C_MSG_STOP;
-
-		if (cnt > sizeof(msg->header.raw_value)) {
-			rv = i2c_transfer(cfg->bus.bus, buf, 3, cfg->bus.addr);
-		} else {
-			buf[1].flags |= I2C_MSG_STOP;
-			rv = i2c_transfer(cfg->bus.bus, buf, 2, cfg->bus.addr);
-		}
-
-		/* If tcpc write fails, return error */
-		if (rv) {
-			return rv;
-		}
-	}
-
-	/*
-	 * We always retry in TCPC hardware since the TCPM is too slow to
-	 * respond within tRetry (~195 usec).
-	 *
-	 * The retry count used is dependent on the maximum PD revision
-	 * supported at build time.
-	 */
-	rv = tcpci_write_reg8(&cfg->bus, TCPC_REG_TRANSMIT,
-			      TCPC_REG_TRANSMIT_SET_WITH_RETRY(cfg->transmit_retries, msg->type));
-
-	return rv;
+	return tcpci_tcpm_transmit_data(&cfg->bus, msg, cfg->transmit_retries);
 }
 
 int ps8xxx_tcpc_dump_std_reg(const struct device *dev)
 {
 	const struct ps8xxx_cfg *cfg = dev->config;
-	uint16_t value;
 
 	LOG_INF("TCPC %s:%s registers:", cfg->bus.bus->name, dev->name);
-	for (unsigned int a = 0; a < TCPCI_STD_REGS_SIZE; a++) {
-		switch (tcpci_std_regs[a].size) {
-		case 1:
-			tcpci_read_reg8(&cfg->bus, tcpci_std_regs[a].addr, (uint8_t *)&value);
-			LOG_INF("- %-30s(0x%02x) =   0x%02x", tcpci_std_regs[a].name,
-				tcpci_std_regs[a].addr, (uint8_t)value);
-			break;
-		case 2:
-			tcpci_read_reg16(&cfg->bus, tcpci_std_regs[a].addr, &value);
-			LOG_INF("- %-30s(0x%02x) = 0x%04x", tcpci_std_regs[a].name,
-				tcpci_std_regs[a].addr, value);
-			break;
-		}
-	}
 
-	return 0;
+	return tcpci_tcpm_dump_std_reg(&cfg->bus);
 }
 
 void ps8xxx_tcpc_alert_handler_cb(const struct device *dev, void *data, enum tcpc_alert alert)
 {
 }
 
 int ps8xxx_tcpc_get_status_register(const struct device *dev, enum tcpc_status_reg reg,
-				    int32_t *status)
+				    uint32_t *status)
```

Comment:
This API was fixed on https://github.com/zephyrproject-rtos/zephyr/pull/82626, so you need to rebase your PR.

Reply:
Done, thanks.